components/raftstore/src/store/fsm/apply.rs

```diff
@@ -1029,15 +1029,19 @@ where
                 break;
             }
 
-            let expect_index = self.apply_state.get_applied_index() + 1;
-            if expect_index != entry.get_index() {
-                panic!(
-                    "{} expect index {}, but got {}, ctx {}",
-                    self.tag,
-                    expect_index,
-                    entry.get_index(),
-                    apply_ctx.tag,
-                );
+            let peer = find_peer_by_id(&self.region, self.id).unwrap();
```

Comment:
How about caching peer or `is_witness` in `ApplyDelegate`? So it doesn't need to search again in every loop.

Reply:
done